### PR TITLE
Challenger pays challenge deposit upfront (to discourage spam)

### PR DESCRIPTION
**Pull request by ewilz**

### CODE REVIEW

**Comment by ewilz** [comment]
@mikec wondering if we should take this opportunity to try and consolidate challenge functions `start()` and `fund()`

**Comment by mikec** [comment]
@emaG3m yah, might as well remove the `fund()` function now

**Comment by ewilz** [comment]
@mikec removed `fund()`

File: contracts/Registry.sol [hunk 1]
```diff
@@ -193,6 +194,8 @@ contract Registry {
         challenges[challengeNonce].listingHash = _listingHash;
         listing.challengeID = challengeNonce;
 
+        token.transfer(challengeAddress, minDeposit);
```
Comment (mikec) [inline contracts/Registry.sol]: @emaG3m may want to wrap this in a `require`. Also, might make sense to move the `transferFrom` line directly above this one, just to group the value transferring logic together

**Comment by ewilz** [comment]
@mikec addressed your comment.

**Comment by mikec** [comment]
👍 🎉

**Pull request merged**